packages/webpack-cli/lib/groups/ConfigGroup.js

```diff
@@ -167,22 +170,26 @@ class ConfigGroup extends GroupHelper {
         if (Object.keys(this.args).some((arg) => arg === 'merge')) {
             const { merge } = this.args;
 
-            const newConfigPath = this.resolveFilePath(merge, 'webpack.base.js');
-            if (newConfigPath) {
-                const configFiles = getConfigInfoFromFileName(newConfigPath);
-                if (!configFiles.length) {
-                    this.opts.processingMessageBuffer.push({
-                        lvl: 'warn',
-                        msg: 'Could not find file to merge configuration with...',
-                    });
-                    return;
-                }
-                const foundConfig = configFiles[0];
-                const resolvedConfig = this.requireConfig(foundConfig);
-                const newConfigurationsObject = await this.finalize(resolvedConfig);
-                const webpackMerge = require('webpack-merge');
-                this.opts['options'] = webpackMerge(this.opts['options'], newConfigurationsObject.options);
+            // Try to resolve the given merge config else fallback to `webpack.base.js`
+            const newConfigPath = this.resolveFilePath(merge);
+
+            if (!newConfigPath) {
+                logger.error("The supplied merge config doesn't exist.");
+                process.exit(1);
+            }
+
+            const configFiles = getConfigInfoFromFileName(newConfigPath);
+            if (!configFiles.length) {
+                this.opts.processingMessageBuffer.push({
+                    lvl: 'warn',
+                    msg: 'Could not find file to merge configuration with...',
+                });
+                return;
```

Comment:
Can you clarify? What means `configFiles.length`? Maybe we need an error here too?

Reply:
Good catch, this is redundant, already throwing above.